common/fee_states.c

```diff
@@ -73,6 +73,16 @@ u32 get_feerate(const struct fee_states *fee_states,
 	abort();
 }
 
+/* Are feerates all agreed by both sides? */
+bool feerate_changes_done(const struct fee_states *fee_states)
+{
+	size_t num_feerates = 0;
+	for (size_t i = 0; i < ARRAY_SIZE(fee_states->feerate); i++) {
+		num_feerates += (fee_states->feerate[i] != NULL);
+	}
+	return num_feerates == 1;
```

Comment:
Is it really necessary to iterate over the entire array? At the very least can't we short-circuit the loop if there is more than one non-NULL entry?

    if (num_feerates == 0 && fee_states->feerate[i] != NULL)
        num_feerates = 1;
    else if (num_feerates == 1 && fee_states->feerate[i] != NULL)
        return false;
    return true

Reply:
I prefer the simpler version, TBH; it's pretty marginal for a 16-entry array...

Reply:
OK fair enough. 16 entries is no big deal; I didn't realize it was so small.

It's just nitpicking, but IMO a pure Boolean logic implementation is simpler and clearer. Also it avoids the risk of integer overflow (not going to happen with 16 elements but generally speaking):
```
bool has_inflight_feerate = false;
// ...
for (size_t i = 0; i < ARRAY_SIZE(fee_states->feerate); i++) {
    if (fee_states->feerate[i] != NULL) {
        if (!has_inflight_feerate)
            has_inflight_feerate = true;
        else
            return false;
    }
}
return true;
```